### PR TITLE
solution first draft

### PR DESCRIPTION
**Pull request by scottcreynolds**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,17 +7,147 @@
...
+applications, we often need to use other HTTP verbs, such as `POST`, to
+write data as well as read it. Luckily, it's very easy to `POST` with
+`fetch` as well.
+
+Let's look at an example of posting a new comment to a commit with the
+Github API. Replace the commit with a commit from one of your
+repositories, and use your token if you want to try this out.
+
+```js
+const token = "YOUR_TOKEN_HERE"
```
Comment (pletcher) [inline README.md:85]: Would it be too much to have students put their token in a `.env` file (which we can add to the .gitignore on our end) and use `dotenv()` to load it? https://www.npmjs.com/package/dotenv

File: README.md [hunk 2]
```diff
@@ -7,17 +7,147 @@
...
+
+Finally, we can examine the response in our `then` function just the
+same as we did with a `GET` request.
+
+**Top-tip:** Make sure you read the API documentation carefully! They
+will often specific which fields are required and which are optional,
+as well as the format of the request body. Github expects JSON data in
+the body, but another API might want form data (which you can create
+with `new FormData()` or XML or something else. Always read the docs!
+expects JSON
```
Comment (pletcher) [inline README.md:116]: Looks like this is a stray

File: test/index-test.js [hunk 3]
```diff
@@ -1,14 +1,99 @@
...
+  })
+
+  describe('fetch functions', () => {
+    let fetchSpy
+    before(() => {
+      window.fetch = require('node-fetch')
+    })
+
+    beforeEach(() => {
+      fetchSpy = expect.spyOn(window, "fetch").andReturn(new Promise(() => {}))
```
Comment (pletcher) [inline test/index-test.js:62]: 👍

File: README.md [hunk 4]
```diff
@@ -7,17 +7,147 @@
...
+authorization.
+
+All of these additional settings go in that `options` argument, which is
+just an object that we can pass as the second argument to `fetch`.
+
+Finally, we can examine the response in our `then` function just the
+same as we did with a `GET` request.
+
+**Top-tip:** Make sure you read the API documentation carefully! They
+will often specific which fields are required and which are optional,
```
Comment (scottcreynolds) [inline README.md:112]: also a typo here. will fix

**Pull request merged**